pkg/proctree/process_context.go

```diff
@@ -29,30 +28,23 @@ type ProcessCtx struct {
 }
 
 type ProcessTree struct {
-	processTreeMap map[int]ProcessCtx
+	ProcessTreeMap map[int]ProcessCtx
 }
 
-func (t *Tracee) ParseProcessContext(ctx []byte) (ProcessCtx, error) {
+func ParseProcessContext(ctx []byte, containers *containers.Containers) (ProcessCtx, error) {
 	var procCtx = ProcessCtx{}
 	procCtx.StartTime = int(binary.LittleEndian.Uint64(ctx[0:8]))
 	cgroupId := binary.LittleEndian.Uint64(ctx[8:16])
-	procCtx.ContainerID = t.containers.GetCgroupInfo(cgroupId).ContainerId
-	decoder := bufferdecoder.New(ctx[16:]) // this is the offset after the cgroup and startTime in the ctx byte array
-	var errs []error
-	errs = append(errs, decoder.DecodeUint32(&procCtx.Pid))
-	errs = append(errs, decoder.DecodeUint32(&procCtx.Tid))
-	errs = append(errs, decoder.DecodeUint32(&procCtx.Ppid))
-	errs = append(errs, decoder.DecodeUint32(&procCtx.HostTid))
-	errs = append(errs, decoder.DecodeUint32(&procCtx.HostPid))
-	errs = append(errs, decoder.DecodeUint32(&procCtx.HostPpid))
-	errs = append(errs, decoder.DecodeUint32(&procCtx.Uid))
-	errs = append(errs, decoder.DecodeUint32(&procCtx.MntId))
-	errs = append(errs, decoder.DecodeUint32(&procCtx.PidId))
-	for _, e := range errs {
-		if e != nil {
-			return procCtx, e
-		}
-	}
+	procCtx.ContainerID = containers.GetCgroupInfo(cgroupId).ContainerId
```

Comment:
Please verify that ctx byte array has at least 52 bytes. This is instead of the previous error checking we had